### PR TITLE
feat: TE7-T1022: Implement Navigation setup

### PR DESCRIPTION
**Pull request by jigar-s-simform**
Implement Navigation setup for Dashboard, login flow, and Drawer screens

### CODE REVIEW

File: app/App.tsx [hunk 1]
```diff
@@ -1,7 +1,13 @@
-import { LoginScreen } from './modules';
+import { NavigationContainer } from '@react-navigation/native';
+import { RootMain } from './navigation';
 
 const App = () => {
```
Comment (sagar-simformsolutions) [inline app/App.tsx]: add return type

File: app/constants/Strings.ts [hunk 2]
```diff
@@ -42,3 +42,18 @@ export default {
     profile: 'Profile',
   },
 };
+export enum NavigationRoutes {
+  SplashScreen = 'SPLASHSCREEN',
```
Comment (sagar-simformsolutions) [inline app/constants/Strings.ts]: use underscore for example SPLASH_SCREEN , apply on others

File: app/modules/home/HomeScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Text, View } from 'react-native';
+
+const HomeScreen = () => {
```
Comment (sagar-simformsolutions) [inline app/modules/home/HomeScreen.tsx]: add types

File: app/modules/home/HomeScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    
+  return (
+    <View>
```
Comment (sagar-simformsolutions) [inline app/modules/home/HomeScreen.tsx]: use <>  and  < / >, if there is no props

File: app/modules/home/HomeScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+      <Text>HomeScreen</Text>
```
Comment (sagar-simformsolutions) [inline app/modules/home/HomeScreen.tsx]: use string file

File: app/modules/login/useLogin.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import { LoginSchemaTypes, loginSchema } from '../../utils';
+
+/**
+ * useLogin Hook
+ *
+ * @description This hook provides the login functionality and handles the login process.
+ *
+ * @returns {Object} An object containing the formik object, loginStatus, formRefs, and navigation.
+ */
+const useLogin = () => {
```
Comment (sagar-simformsolutions) [inline app/modules/login/useLogin.ts]: add types

File: app/modules/profile/ProfileScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import { Text, View } from 'react-native'
+
+const ProfileScreen = () => {
```
Comment (sagar-simformsolutions) [inline app/modules/profile/ProfileScreen.tsx]: add types

File: app/modules/profile/ProfileScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    
+  return (
+    <View>
```
Comment (sagar-simformsolutions) [inline app/modules/profile/ProfileScreen.tsx]: use < > and < /> if no props are used

File: app/modules/settings/SettingsScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Text, View } from 'react-native';
+
+const SettingsScreen = () => {
```
Comment (sagar-simformsolutions) [inline app/modules/settings/SettingsScreen.tsx]: add types

File: app/modules/settings/SettingsScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+    
+  return (
+    <View>
```
Comment (sagar-simformsolutions) [inline app/modules/settings/SettingsScreen.tsx]: use this tags < > </ >

File: app/navigation/NavigationTypes.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import { NavigationRoutes } from '../constants/strings';
+
+export type RootStackParamsList = {
+  [NavigationRoutes.LoginScreen]: undefined;
```
Comment (sagar-simformsolutions) [inline app/navigation/NavigationTypes.ts:4]: do not provide type as undefined

**Review by sagar-simformsolutions — CHANGES_REQUESTED** [review]

File: app/navigation/NavigationTypes.ts (hunk 11 above)
Comment (jigar-s-simform) [reply app/navigation/NavigationTypes.ts:4]: If screen is not getting any params then we have to specify as undefined!
It is mentioned in the Documentation of react-navigation to specify undefined.

please check this link of documentation: [Visit documentation](https://reactnavigation.org/docs/typescript)

File: app/constants/Strings.ts [hunk 12]
```diff
@@ -42,3 +42,4 @@ export default {
     profile: 'Profile',
   },
 };
+
```
Comment (sagar-simformsolutions) [inline app/constants/Strings.ts]: remove unnecessary spaces

**Review by sagar-simformsolutions — CHANGES_REQUESTED** [review]

File: app/constants/Strings.ts (hunk 12 above)
Comment (jigar-s-simform) [reply app/constants/Strings.ts]: resolved

File: app/navigation/NavigationTypes.ts (hunk 11 above)
Comment (jigar-s-simform) [reply app/navigation/NavigationTypes.ts:4]: resolved

File: app/modules/settings/SettingsScreen.tsx (hunk 10 above)
Comment (jigar-s-simform) [reply app/modules/settings/SettingsScreen.tsx]: resolved

File: app/modules/settings/SettingsScreen.tsx (hunk 9 above)
Comment (jigar-s-simform) [reply app/modules/settings/SettingsScreen.tsx]: resolved

File: app/modules/profile/ProfileScreen.tsx (hunk 8 above)
Comment (jigar-s-simform) [reply app/modules/profile/ProfileScreen.tsx]: resolved

File: app/modules/profile/ProfileScreen.tsx (hunk 7 above)
Comment (jigar-s-simform) [reply app/modules/profile/ProfileScreen.tsx]: resolved

File: app/modules/home/HomeScreen.tsx (hunk 5 above)
Comment (jigar-s-simform) [reply app/modules/home/HomeScreen.tsx]: resolved

File: app/modules/login/useLogin.ts (hunk 6 above)
Comment (jigar-s-simform) [reply app/modules/login/useLogin.ts]: resolved

File: app/modules/home/HomeScreen.tsx (hunk 4 above)
Comment (jigar-s-simform) [reply app/modules/home/HomeScreen.tsx]: resolved

File: app/modules/home/HomeScreen.tsx (hunk 3 above)
Comment (jigar-s-simform) [reply app/modules/home/HomeScreen.tsx]: resolved

File: app/constants/Strings.ts (hunk 2 above)
Comment (jigar-s-simform) [reply app/constants/Strings.ts]: resolved

File: app/App.tsx (hunk 1 above)
Comment (jigar-s-simform) [reply app/App.tsx]: resolved

**Review by sagar-simformsolutions — APPROVED** [review]

**Pull request merged**